### ISSUE TITLE
RESTful API for database

### ISSUE THREAD

**Issue by jsws** [ISISScientificComputing/autoreduce#73]
Issue raised by: developer

### What?
It would be nice to have a RESTful API for autoreduction.

Currently we have two places where the database is modified, in the web app when a job is reran and in the queue processor. This doesn't make much sense because we have to have a copy of the models in the queue processor, as well as the messiness of having two places do the same thing. An API will provide a layer of abstraction, with everything in the database being updated in a single place.

For example, when submitting a job you'd just have to make a HTTP request to `/api/submit` with the instrument and run number in request body. The API would then handle creation of the reduction run in the database and adding the reduction run to a queue to be reduced. Then once the reduction is complete another call is made to the API. 

Having an API would allow us to use any front end framework, such as making autoreduction a Topcat plugin, meaning scientists only have to go to one website.

### How to test the issue is resolved
1. Ensure we have a good comprehensive design for this.
2. Identify areas of the code base that would need to be changed to accommodate the new communication method
3. Implement - probably in smaller PRs/issues
4. Test manually rigorously

**Comment by ElliotAOram** [comment]
Will be handled differently by database refactoring. However this is the kind of line we will be following

**Comment by ElliotAOram** [comment]
Re-opening as relevant again

**Comment by ElliotAOram** [comment]
Django REST framework: https://www.django-rest-framework.org/

**Comment by dtasev** [comment]
It would be nice but I don't see it happening within the next year or more, and not that necessary IMO
